main.c

```diff
@@ -234,13 +273,15 @@ static void keyboard_handle_key(void *data, struct wl_keyboard *wl_keyboard,
 	case WL_KEYBOARD_KEY_STATE_PRESSED:
 		switch (keysym) {
 		case XKB_KEY_Escape:
-			seat->has_selection = false;
+			seat->pointer_selection.has_selection = false;
+			seat->touch_selection.has_selection = false;
 			state->edit_anchor = false;
 			state->running = false;
 			break;
 
 		case XKB_KEY_space:
-			if (!seat->has_selection) {
+			if (!seat->pointer_selection.has_selection &&
+			    !seat->touch_selection.has_selection) {
```

Comment:
Style: we don't alignm, we indent with two tabs instead (to make a clear difference between the if condition and block).